pkg/instancetype/preference/requirements/memory.go

```diff
@@ -37,17 +40,23 @@ func checkMemory(
 	preferenceSpec *v1beta1.VirtualMachinePreferenceSpec,
 	vmiSpec *virtv1.VirtualMachineInstanceSpec,
 ) (conflict.Conflicts, error) {
-	if instancetypeSpec != nil && instancetypeSpec.Memory.Guest.Cmp(preferenceSpec.Requirements.Memory.Guest) < 0 {
-		instancetypeMemory := instancetypeSpec.Memory.Guest.String()
-		preferenceMemory := preferenceSpec.Requirements.Memory.Guest.String()
-		return conflict.Conflicts{conflict.New("spec", "instancetype")},
-			fmt.Errorf(InsufficientInstanceTypeMemoryResourcesErrorFmt, instancetypeMemory, preferenceMemory)
+	errFmt := InsufficientVMMemoryResourcesErrorFmt
+	errConflict := conflict.New("spec", "template", "spec", "domain", "memory")
+	providedMemory := pointer.P(resource.MustParse("0Mi"))
+
+	if instancetypeSpec != nil {
+		errConflict = conflict.New("spec", "instancetype")
+		errFmt = InsufficientInstanceTypeMemoryResourcesErrorFmt
+		providedMemory = &instancetypeSpec.Memory.Guest
```

Comment:
Silly question, it is possible to specify a instanceType without `Memory.guest`? If possible, we should check if this is nil?

Reply:
afaik spec.Memory.guest is mandatory

Reply:
Yeah `spec.{cpu,memory}.guest` are both required within an instance type at the moment.

Reply:
Thanks for the clarifications!